server/neptune/workflows/terraform.go

```diff
@@ -16,8 +20,11 @@ type TerraformActivities struct {
 	activities.Terraform
 }
 
-func NewTerraformActivities(scope tally.Scope, serverURL *url.URL) (*TerraformActivities, error) {
-	terraformActivities := activities.NewTerraform(serverURL)
+func NewTerraformActivities(config neptune.TerraformConfig, dataDir string, scope tally.Scope, serverURL *url.URL, logger logging.Logger) (*TerraformActivities, error) {
```

Comment:
ah i see why you moved TerraformConfig, but i would just move that here instead of keeping it in `neptune/config`

Reply:
I don't think we need to pass scope and logger into our activities, for scope we have:
https://github.com/temporalio/samples-go/blob/main/metrics/workflow.go#L36

and for logging we have:
https://github.com/lyft/atlantis/blob/25d3a061e24f5658a70889d9eed73d1cb81bbc6b/server/neptune/workflows/internal/activities/config/logger/logger.go#L10-L11

Reply:
Gotcha, we'd still need to build the metric name similar to how we build scope right? Also, I have wrapped the logging methods in an interface so that we can mock it for tests without having to worry about workflow and activity contexts. What do you think?

Reply:
Not really a fan of this, we went from not needing to inject our logger at all to injecting it.  The struct doesn't have any state that we need to maintain at all either.

I'd rather you see what other folks using temporal are doing first before needing to do this (ie. by asking in the OSS channel, or looking at samples).